.github/workflows/gbs_x64.yml

```diff
@@ -10,30 +10,76 @@ jobs:
     runs-on: ubuntu-20.04
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Check if rebuild required
+      ## @todo This should become a reusable workflow.
+      run: |
+        tmpfile=$(mktemp)
+        git show --pretty="format:" --name-only --diff-filter=AMRC ${{ github.event.pull_request.head.sha}} -${{ github.event.pull_request.commits }} | sort | uniq | awk NF > ${tmpfile}
+        echo "{changed_file_list}={${tmpfile}}" >> "$GITHUB_ENV"
+        rebuild='0'
+        regbs='0'
+        redebian='0'
+        reandroid='0'
+        for file in `cat ${tmpfile}`; do
+          echo "Checking: ${file}."
+          case $file in
+            *.md|*.png|*.webp|*.css|*.html )
+              echo "nochg"
+              ;;
+            packaging/* )
+              regbs='1'
+              ;;
+            debian/* )
+              redebian='1'
+              ;;
+            jni/* )
+              reandroid='1'
+              ;;
+            * )
+              rebuild='1'
+              regbs='1'
+              redebian='1'
+              reandroid='1'
+              ;;
+          esac
+        done
+        echo "Rebuild required: ${rebuild}"
+        echo "rebuild=${rebuild}" >> "$GITHUB_ENV"
+        echo "GBS required: ${regbs}"
+        echo "regbs=${regbs}" >> "$GITHUB_ENV"
+        echo "PDEBUILD required: ${redebian}"
+        echo "redebian=${redebian}" >> "$GITHUB_ENV"
+        echo "JNI required: ${reandroid}"
+        echo "reandroid=${reandroid}" >> "$GITHUB_ENV"
     - uses: actions/setup-python@v1
-    - name: prepare deb sources for GBS
-      run: echo "deb [trusted=yes] http://download.tizen.org/tools/latest-release/Ubuntu_20.04/ /" | sudo tee /etc/apt/sources.list.d/tizen.list
-    - name: install GBS
-      run: sudo apt-get update && sudo apt-get install -y gbs
-    - name: configure GBS
-      run: cp .github/workflows/tizen.gbs.conf ~/.gbs.conf
-    - name: make cache key
-      id: make-key
-      run: echo "cache_key=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
-      shell: bash
+    - name: prepare GBS
+      if: env.regbs == '1'
+      run: |
+        echo "deb [trusted=yes] http://download.tizen.org/tools/latest-release/Ubuntu_20.04/ /" | sudo tee /etc/apt/sources.list.d/tizen.list
+        sudo apt-get update && sudo apt-get install -y gbs
+        cp .github/workflows/tizen.gbs.conf ~/.gbs.conf
+        echo "cache_key=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
```

Comment:
`"run GBS"` needs key of id `make-key`. How about keep `"make cache key"` part with `if: env.regbs`

Reply:
applied. thanks!